Containerfile

```diff
@@ -0,0 +1,22 @@
+FROM registry.fedoraproject.org/fedora:36
+LABEL maintainer="Red Hat"
+
+WORKDIR /src
+RUN dnf -y install \
+    --setopt=deltarpm=0 \
+    --setopt=install_weak_deps=false \
+    --setopt=tsflags=nodocs \
+    golang \
+    git-core \
+    python3-devel \
+    python3-pip \
+    && dnf clean all
+
+COPY . .
+
+# Mount the .git folder so that setuptools_scm can resolve the version
+RUN --mount=type=bind,source=.git,target=/src/.git \
```

Comment:
Doesn't the `COPY . .` already copy everything including .git? Should we have a containerignore that excludes things, or copy only the things we need?

Reply:
Oh! I didn't notice we had one already!

Could we add `.mypy_cache/` to the ignore file? Out of the missing ignores, that seems to be the biggest one, 15.3 MiB on my laptop

+:100: for mounting the .git directory

Reply:
Does it work for you? I get a setuptools-scm error :confused:

```
  LookupError: setuptools-scm was unable to detect version for /src.
  
  Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.
```

Reply:
I think it's a SELinux permissions things, adding `Z` worked (the same `Z` used with `--volume=src:dest:Z`
```
RUN --mount=type=bind,source=.git,target=/src/.git,Z \
```

Reply:
So, as we did in https://github.com/containerbuildsystem/cachito/pull/742, I have removed the mount in favor of copying the .git folder to the image.